pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -136,10 +136,13 @@ func (r *ReplicaSetReconciler) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
-	r.log.Debug("Building service")
-	svc := buildService(mdb)
-	if err = r.client.CreateOrUpdate(&svc); err != nil {
-		r.log.Infof("The service already exists... moving forward: %s", err)
+	r.log.Debug("Building and creating service")
+	if err = r.createService(mdb); err != nil {
+		if errors.IsAlreadyExists(err) {
+			r.log.Infof("The service already exists... moving forward: %s", err)
+		} else {
+			return reconcile.Result{}, err
```

Comment:
One thing to keep in mind here, returning `reconcile.Result{}, err` will requeue a new reconcilliation, but we aren't actually getting any info about this error. We should make sure to log it before hand.